pkg/controllermanager/controller/controllerregistration/seed/add.go

```diff
@@ -124,7 +124,8 @@ func (r *Reconciler) SeedPredicate() predicate.Predicate {
 			}
 
 			return !apiequality.Semantic.DeepEqual(oldSeed.Spec.DNS.Provider, seed.Spec.DNS.Provider) ||
-				seed.DeletionTimestamp != nil
+				seed.DeletionTimestamp != nil ||
+				topologyAwareRoutingSettingsHasChanged(oldSeed.Spec.Settings, seed.Spec.Settings)
```

Comment:
While this fixes the specific problem, I think we should rather change the predicate such that it returns `true` whenever
- deletion timestamp is set
- labels change
- annotations change
- spec changes

The reason is that we provide this information when installing the Helm charts https://github.com/gardener/gardener/blob/1278a323d19c5aaae0eed120024960beefa685fe/pkg/gardenlet/controller/controllerinstallation/controllerinstallation/reconciler.go#L201-L218

Hence, any extension could rely on any field in these structs, just like it happened now for the topology-aware routing setting.